### PR TITLE
Use a different fee currency if there's not enough to pay for gas

### LINKED ISSUE #2231: Make sure users have enough balance to pay for gas before sending txs.
In the places we (or other dapp developers) don't set a feeCurrency, the app tries to use CELO if the user has enough balance. The problem is that  we check for > 0, not for > gasCost, so if the user has a very small amount of CELO (less than needed to send a tx) then we still try to use CELO and the tx fails.

This has already been affecting some users: https://sentry.io/organizations/valora-inc/issues/3004252806/?project=1250733&query=is%3Aunresolved+gas and the only workaround is to either get rid of their whole CELO balance or getting more CELO.

We should only use a feeCurrency if the user has enough to pay for gas with it.

### PR DESCRIPTION
**Pull request by gnardini**
### Description

In the places we (or other dapp developers) don't set a feeCurrency, the app tries to use CELO if the user has enough balance. The problem is that  we check for > 0, not for > gasCost, so if the user has a very small amount of CELO (less than needed to send a tx) then we still try to use CELO and the tx fails.

This has already been affecting some users: https://sentry.io/organizations/valora-inc/issues/3004252806/?project=1250733&query=is%3Aunresolved+gas and the only workaround is to either get rid of their whole CELO balance or getting more CELO.

We are now only using a feeCurrency if the user has enough to pay for gas with it.

### Other changes

N/A

### Tested

With unit tests and manually

### How others should test

- Get an account with a very small amount of CELO (<0.000001)
- Invite someone using the Send flow.
- Try to reclaim the invite. Reclaiming should be successful.

### Related issues

- Fixes #2231

### Backwards compatibility

N/A

### CODE REVIEW

**Review by jeanregisser — APPROVED** [review]
Amazing!! 💯 👍

**Comment by MuckT** [comment]
@jeanregisser @gnardini would be great to apply this as a patch as 1.29.1. Will Cut 1.29.0 and we can work to include this over the next few days.

File: packages/mobile/src/walletConnect/request.test.ts [hunk 1]
```diff
@@ -206,7 +206,7 @@ describe(handleRequest, () => {
             data: '0xABC',
             feeCurrency: undefined, // undefined to pay with CELO, since the balance is non zero
             gas: 1,
-            gasPrice: 3,
```
[the remark this answers is not in the document]
Comment (gnardini) [reply packages/mobile/src/walletConnect/request.test.ts:209]: Alright, from the comment above the `feeCurrency` could have been specified as something different but then stripped by WC so we should not trust the value of `gasPrice` that we receive. Removed that so we re-calculate it. The mock we're using is a different one so the value still changed but we're not using the one passed by WC.

I misunderstood the expected behavior of this test, my bad.

**Comment by ValoraQA** [comment]
Hey @MuckT, We have verified above task on Latest Test Flight Release build V 1.30.1 and updated status in test rails.

**Observation:**
- User is able to invite new number and able to reclaim the invite sent from by tapping on "Reclaim" button from the invite cta for the account have very small amount of CELO.

**Test rail link:**
**iOS15:** https://valoraapp.testrail.io/index.php?/tests/view/27223

**Device:** iPhone 13 mini(15.1.1)

**CC:** @gnardini 

Thanks..!!

**Comment by ValoraQA** [comment]
Hey @MuckT, We have verified above task on Latest Test Flight Release build V 1.30.1 and updated status in test rails.

Observation:

User is able to invite new number and able to reclaim the invite sent from by tapping on "Reclaim" button from the invite CTA for the account have very small amount of CELO & activity with "Unknown" title appears in recent activity section.

Test rail link:
iOS14: https://valoraapp.testrail.io/index.php?/tests/view/27316

Device: iPhone 12 (14.7.1)

CC: @gnardini
Thanks..!!

**Pull request merged**